syntax="proto3";

// 包名是生成代码的使用的namespace，所有代码都在这个下面
package Experiment;

// 指定服务的名称，作为生成代码里面的二级namespace
service Server {
  rpc Test(Request) returns (Reply) {}
  // 测试接口四 输出流
  rpc TestOutStream(Request) returns(stream Reply){}
  // 测试接口五 输入流
  rpc TestInStream(stream Request) returns(Reply){}
  // 测试接口六 双向流
  rpc TestBidirectionalStream(stream Request) returns(stream Reply){}
}


message Request {
  string name   = 1; // 客户端名称
  int32  id     = 2; // 客户端IP
  double value  = 3; // 附带一个值
}

message Reply {
  uint64 tid        = 1; // 服务线程ID
  string svrname    = 2; // 服务名称
  double takeuptime = 3; // 请求处理耗时
  string extend     = 4; // 额外的
}